### PR TITLE
Simplify deprecate command method

### PR DESCRIPTION
**Pull request by bronzdoc**
# Description:
 
Simplify `Gem::Deprecate#deprecate_command` that was introduced here https://github.com/rubygems/rubygems/pull/2935
______________

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: test/rubygems/test_deprecate.rb [hunk 1]
```diff
@@ -95,13 +95,13 @@ def execute
...
   def test_deprecate_command
+    foo = FooCommand.new
+
     out, err = capture_io do
-      foo = FooCommand.new
       foo.execute
     end
 
-    assert_equal "pew pew!\n", out
-    assert_match(/NOTE: foo command is deprecated. It will be removed on or after 2099-04-01.\n/, err)
```
Comment (deivid-rodriguez) [inline test/rubygems/test_deprecate.rb]: Can we keep this assertion, or add a separate test to check it? I'd like the following line to be covered:

https://github.com/rubygems/rubygems/blob/3235920c22a416c95a27f5332e0b39cf72bd6532/lib/rubygems/command_manager.rb#L179

**Comment by duckinator** [comment]
@bronzdoc woah, this is a _great_ improvement, nicely done! After you add the test @deivid-rodriguez mentioned, I think this is good to go!

File: test/rubygems/test_deprecate.rb (hunk 1 above)
Comment (bronzdoc) [reply test/rubygems/test_deprecate.rb]: Done @deivid-rodriguez

**Comment by deivid-rodriguez** [comment]
@bundlerbot r+

**Pull request merged**